syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";
import "graph.proto";

package transport_proto;

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    uint32 id = 1;
    string name = 2;
    Coordinates coord = 3;
}

message Bus {
    uint32 id = 1;
    string name = 2;
    repeated uint32 route = 3;
    uint32 type = 4;
    double route_geo_length = 5;
    double route_true_length = 6;
    uint32 stops_on_route = 7;
    uint32 unique_stops = 8;
}

message RouteSettings {
    double bus_velocity = 1;
    uint32 bus_waiting_time = 2;
}

message TransportCatalogue {
    repeated Stop stop = 1;
    repeated Bus bus = 2;
    MapRenderSettings map_render_setting = 3;
    RouteSettings route_settings = 4;
    Graph graph = 5;
    Router router = 6;
}
